### PR TITLE
save mutating dict while iterating over it

### PR DESCRIPTION
**Pull request by morkovka1337**
This patch fixes unsafe mutating of the dictionary while iterating over it. Previous version of this code lead to incorrect conversion of the keys (not all keys were converted from str to int).

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: tools/accuracy_checker/accuracy_checker/annotation_converters/im2latex.py [hunk 1]
```diff
@@ -40,9 +40,7 @@ def read_vocab(vocab_path):
     elif vocab_path.suffix == '.json':
         with open(vocab_path, "r") as f:
             vocab_dict = json.load(f)
-            for k, v in vocab_dict['id2sign'].items():
-                del vocab_dict['id2sign'][k]
-                vocab_dict['id2sign'][int(k)] = v
+            vocab_dict['id2sign'] = {int(k): v for k, v in vocab_dict['id2sign'].items()}
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/im2latex.py:43]: do you can guarantee that keys are always can be casted to the int?

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/im2latex.py:43]: you still modify vocab during iteration
```suggestion
            id2sign = {int(k): v for k, v in vocab_dict['id2sign'].items()}
            vocab_dict['id2sign'] = id2sign
```
or provide json reading scheme which will cast it automatically

**Review by eaidova — APPROVED** [review]

**Comment by eaidova** [comment]
Jenkins ok to test

**Pull request merged**